File: web-java/src/main/java/pl/maciejcp/web/ScalaJacksonMapper.java
package pl.maciejcp.web;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.module.scala.DefaultScalaModule;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

@Component
public class ScalaJacksonMapper {
    @Bean
    public ObjectMapper buildScalaMapper() {
        ObjectMapper mapper = new ObjectMapper();
        return mapper.registerModule(new DefaultScalaModule());
    }
}

File: web-scala/src/main/scala/pl/maciejcp/web/ScalaJacksonMapper.scala
package pl.maciejcp.web

;

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.scala.DefaultScalaModule
import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Component;

@Component
class ScalaJacksonMapper {
  @Bean
  def buildScalaMapper(): ObjectMapper = {
    val mapper = new ObjectMapper()
    mapper.registerModule(DefaultScalaModule)
  }
}
